redash/models.py

```diff
@@ -658,10 +660,9 @@ def all_queries(cls, groups, drafts=False):
             .order_by(cls.created_at.desc())
 
         if drafts:
-            q = q.where(Query.name == 'New Query')
+            q = q.where(Query.is_draft == True)
         else:
-            q = q.where(Query.name != 'New Query')
-
+            q = q.where(Query.is_draft == False)
```

Comment:
If you decide you want to implement #3, then I prefer we reduce the scope of this pull request to adding the new field and maintain current behavior (i.e. implement #2). And then in a future PR, implement the new search/recents logic. I think the search change might take time, and I want to avoid having a long living pull request hanging around.

And of course, we can implement #3 in this pull request and decide later on if we want to "upgrade" to #1.